ambry-store/src/main/java/com.github.ambry.store/BlobStoreCompactor.java

```diff
@@ -750,6 +756,51 @@ private boolean isDuplicate(IndexEntry copyCandidate, FileSpan duplicateSearchSp
     return validEntries;
   }
 
+  /**
+   * Gets the {@link IndexValue} for the PUT from the {@link #srcIndex} (if it exists)
+   * @param key the {@link StoreKey} whose PUT is required
+   * @param updateValue the update (TTL update/delete) {@link IndexValue} associated with the same {@code key}
+   * @param indexSegmentOfUpdateValue the {@link IndexSegment} that {@code updateValue} belongs to
+   * @return the {@link IndexValue} for the PUT in the {@link #srcIndex} (if it exists)
+   * @throws StoreException if there are problems with the index
+   */
+  private IndexValue getPutValueFromSrc(StoreKey key, IndexValue updateValue, IndexSegment indexSegmentOfUpdateValue)
+      throws StoreException {
+    IndexValue putValue = srcIndex.findKey(key, new FileSpan(srcIndex.getStartOffset(), updateValue.getOffset()),
+        EnumSet.of(PersistentIndex.IndexEntryType.PUT));
+    // in a non multi valued segment, there can only be PUTs and DELETEs in the same segment
```

Comment:
The comment is a little misleading/hard to understand. How about:
In a non multi valued segment, if putValue is not found in previous indexSegments, check if PUTs and DELETEs in the same segment.

Reply:
Changed wording. Let me know what you think